src/AgentBuilder.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+
+namespace Nipwaayoni;
+
+use Http\Discovery\HttpClientDiscovery;
+use Http\Discovery\Psr17FactoryDiscovery;
+use Nipwaayoni\Contexts\ContextCollection;
+use Nipwaayoni\Events\DefaultEventFactory;
+use Nipwaayoni\Events\EventFactoryInterface;
+use Nipwaayoni\Config;
+use Nipwaayoni\Stores\TransactionsStore;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+
+class AgentBuilder
+{
+    /** @var Config */
+    private $config;
+
+    /** @var array */
+    private $sharedContexts;
+
+    /** @var array */
+    private $tags;
+
+    /** @var array */
+    private $env;
+
+    /** @var array */
+    private $cookies;
+
+    /** @var EventFactoryInterface */
+    private $eventFactory;
+
+    /** @var TransactionsStore */
+    private $transactionStore;
+
+    /** @var ClientInterface */
+    private $httpClient;
+
+    /** @var RequestFactoryInterface */
+    private $requestFactory;
+
+    /** @var StreamFactoryInterface */
+    private $streamFactory;
+
+    public function __construct()
+    {
+        $this->init();
+    }
+
+    private function init(): void
+    {
+        $this->config = new Config(['appName' => 'APM Agent']);
+
+        $this->sharedContexts = [
+            'user' => [],
+            'custom' => [],
+        ];
+
+        $this->tags = [];
+
+        $this->env = [];
+
+        $this->cookies = [];
+
+        $this->eventFactory = new DefaultEventFactory();
+        $this->transactionStore = new TransactionsStore();
+        $this->httpClient = HttpClientDiscovery::find();
+        $this->requestFactory = Psr17FactoryDiscovery::findRequestFactory();
+        $this->streamFactory = Psr17FactoryDiscovery::findStreamFactory();
+    }
+
+    public function make(): Agent
```

Comment:
Hi @dstepe, to keep names concise I'd suggest to change this method to `build` as the class is a "builder". The same approach could be replied to the others `make*` methods.

Reply:
Changed.